packages/demo-wallet-client/src/ducks/allAssets.ts

```diff
@@ -15,43 +21,62 @@ import {
 
 type IncludeAssetOverridesProps = {
   assets: Asset[];
+  balances?: Types.BalanceMap;
   assetCategory: AssetCategory;
   assetOverrides: Asset[];
 };
 
 const includeAssetOverrides = ({
   assets,
+  balances,
   assetCategory,
   assetOverrides,
-}: IncludeAssetOverridesProps) =>
-  assets.reduce((result: Asset[], asset) => {
-    const overrideAsset = assetOverrides.find(
-      (ao) => ao.assetString === asset.assetString,
-    );
-    const updatedAsset = overrideAsset
-      ? {
-          ...overrideAsset,
-          category: assetCategory,
-          isUntrusted: asset.isUntrusted,
-          isOverride: true,
-          total: asset.total,
-        }
-      : { ...asset, category: assetCategory };
-
-    return [...result, updatedAsset];
-  }, []);
+}: IncludeAssetOverridesProps) => {
+  const isUntrusted = assetCategory === AssetCategory.UNTRUSTED;
+  const overrides = assetOverrides.filter((o) => o.isUntrusted === isUntrusted);
+
+  return [...assets, ...overrides].map((a) => ({
+    ...a,
+    category: assetCategory,
+    // Use balance from account balances
+    total: balances?.[a.assetString]?.available?.toString() || a.total || "0",
+  }));
+};
 
+// All assets (from account balances, assetOverrides, and untrustedAssets) are
+// aggregated here to display on the UI.
+
+// Assets with overrides will be removed from untrustedAssets and will be skipped
+// in account balances to avoid making API calls with on-chain home domain. All
+// assets with overrides will be added to assetOverrides in store and picked from
+// there in this action.
 export const getAllAssetsAction = createAsyncThunk<
   Asset[],
   undefined,
   { rejectValue: RejectMessage; state: RootState }
->("allAssets/getAllAssetsAction", (_, { rejectWithValue, getState }) => {
-  const { assets } = accountSelector(getState());
+>("allAssets/getAllAssetsAction", async (_, { rejectWithValue, getState }) => {
+  const networkConfig = getNetworkConfig();
+
+  const { data } = accountSelector(getState());
   const { data: untrustedAssets } = untrustedAssetsSelector(getState());
   const { data: assetOverrides } = assetOverridesSelector(getState());
+  const { assetOverrides: asssetOverrideSetting } = settingsSelector(
```

Comment:
Using data from the settings (URL) in some places because the store was causing issues due to race conditions (updating multiple ducks simultaneously or in sequence). It could be better, but it's the best we can do to keep the update small.